site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,507 @@
+
+### Step 2: Import multiple service worker modules {: #step-2 }
+
+Our service worker implements two features. For better maintainability, we will implement each feature in a separate module. First, we need to declare the service worker as an [ES Module][mdn-es-module] in our manifest, which allows us to import modules in our service worker:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json/3-3
+{
+ "background": {
+    "service_worker": "service-worker.js",
+    "type": "module"
+  },
+}
+```
+
+Create the `service-worker.js` file and import two modules:
+
+```js
+import './sw-omnibox.js';
+import './sw-tips.js';
+```
+
+Create these files and add a console log to each one.
+
+{% Columns %}
+
+{% Column %}
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+console.log("sw-omnibox.js")
+```
+{% endColumn %}
+
+{% Column %}
+{% Label %}sw-tips.js:{% endLabel %}
+
+```js
+console.log("sw-tips.js")
+```
+{% endColumn %}
+
+{% endColumns %}
+
+See [Importing scripts](tbd) to learn about other ways to import multiple files in a service worker.
+
+
+{% Aside 'important' %}
+
+Remember to set `type.module` when using a modern module bundler framework, such as [CRXjs Vite plugin][crxjs-vite].
+
+{% endAside %}
+
+### _Optional: Debugging the service worker_ {: #step-3 }
+
+Let's quickly go over how to find the service worker logs and know when it has terminated. First, follow the instructions to [Load an unpacked extension][doc-dev-basics-unpacked]. 
+
+After 30 seconds you will see "service worker(inactive)" which means the service worker has terminated. Click on the "service worker(inactive)" hyperlink to inspect it. See example below.
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/D1XRaA6q4xn9Ylwe1u1N.mp4", width="800", height="314", autoplay="true", muted="true", loop="true" %}
+
+Did you notice that inspecting the service worker woke it up? That's right! Opening the service worker in the devtools will keep it active. To make sure that your extension behaves correctly when your service worker is terminated, remember to close the DevTools.
+
+Now let's break the extension to learn where to locate errors. One way to do this is to delete the ".js" from the `'./sw-omnibox.js'` import in the `service-worker.js` file. Chrome will be unable to register the service worker.
+
+Go back to chrome://extensions and refresh the extension. The following error will appear:
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/AbMNDSbURLKjH1Jm1C9Q.mp4", width="400", height="477", autoplay="true", muted="true", loop="true" %}
+
+See [Debugging extensions](tbd) for more ways debug the extension service worker.
+
+{% Aside 'caution' %}
+Don't forget to fix the file name before moving on!
+{% endAside %}
+
+### Step 4: Initialize the state {: #step-4 }
+
+Extensions can save initial values to storage. To use the [`chrome.storage`][api-storage] API, we need to request permission in the manifest:
```

Comment:
```suggestion
Chrome will shut down service workers if they are not needed. We use the [`chrome.storage`][api-storage] API to persist state across service worker sessions. For storage access, we need to request permission in the manifest:
```
I like to give a tiny bit of the why when starting a new step in a tutorial - wdyt?

Reply:
Sounds good :)